scripts/mono-link-ui.js

```diff
@@ -0,0 +1,53 @@
+import fs from 'fs/promises';
+
+const filePath = './package.json';
+const backupFilePath = './package-original.json';
+
+async function modifyUiPackageJson() {
+  try {
+    // Check if a backup already exists
+    try {
+      await fs.access(backupFilePath);
+      console.log(`Backup file already exists: ${backupFilePath}`);
+    } catch {
+      await fs.copyFile(filePath, backupFilePath);
+      console.log(`Backup created: ${backupFilePath}`);
+    }
+
+    // Read and parse the package.json file
+    const data = await fs.readFile(filePath, 'utf-8');
+    let packageJson = JSON.parse(data);
+
+    // Modify the @leather.io/ui dependency
+    packageJson.dependencies['@leather.io/ui'] = 'file:../mono/packages/ui';
+
+    // Ensure pnpm and pnpm.overrides are initialized
+    if (!packageJson.pnpm) {
+      packageJson.pnpm = {};
+    }
+    if (!packageJson.pnpm.overrides) {
+      packageJson.pnpm.overrides = {};
+    }
+
+    // Add the specified overrides
+    const overrides = {
+      '@leather.io/rpc': 'file:../mono/packages/rpc',
+      '@leather.io/constants': 'file:../mono/packages/constants',
+      '@leather.io/models': 'file:../mono/packages/models',
+      '@leather.io/tokens': 'file:../mono/packages/tokens',
+      '@leather.io/utils': 'file:../mono/packages/utils'
```

Comment:
Are these all the overrides needed? I typ override them all, but maybe that is overkill?